src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1110,11 +1110,40 @@ bool entitiesQuery
     }
 
     /* Process query result */
-    while (cursor->more()) {
+    while (cursor->more())
+    {
+        BSONObj                 r    = cursor->next();
+        std::string             err  = r.getStringField("$err");
+        ContextElementResponse* cer  = new ContextElementResponse();
+
+        if (err != "")
+        {
+          //
+          // We can't return the error 'as is', as it may contain forbidden characters.
+          // So, we can just match the error and sent a less descriptive text
+          //
+          const char* invalidPolygon      = "Exterior shell of polygon is invalid";
+          const char* defaultErrorString  = "Internal mongo problem";
+
+          LM_W(("Database Error (%s)", err.c_str()));
+
+          if (strncmp(err.c_str(), invalidPolygon, strlen(invalidPolygon)) == 0)
```

Comment:
What I looked for with that 'not detected at creation time' was to inform that the error isn't here, but was incorrectly inserted that way earlier. But you are right, it may be misinterpreted in more than one way.
I'll remove it.

About changing strings for different versions etc.
Don't worry about that, we just add another string to check against whenever necessary. If the string is not found, we just return the default value (right now: "Internal mongo problem" - which I will change).